File: HelpSoftware/Help/Help.Web/Web/src/app/EF/UserToken.ts
import { User } from "./EF";

export class UserToken {
  ID_UserToken: number;
  User: User;
  Created: Date;
  Token: string;

  deserialize(input) {
    this.ID_UserToken = input.ID_UserToken;
    this.User = new User().deserialize(input.User);
    this.Created = input.Created;
    this.Token = input.Token;
    return this;
  }
}

File: HelpSoftware/Help/Help.EF/Entitys/UserToken.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Help.EF
{
  public class UserToken : IHelpEntity
  {
    [Key]
    public virtual int ID_UserToken { get; set; }

    public virtual User User { get; set; }

    public virtual DateTime Created { get; set; }

    public virtual string Token { get; set; }

    [Timestamp]
    public byte[ ] TimeStamp { get; set; }

    public string GetEntityName()
    {
      return "UserToken";
    }

    public int GetID()
    {
      return ID_UserToken;
    }

    public bool GetWriteHistoryEntry()
    {
      return false;
    }
  }
}